hax/hax/handler.py

```diff
@@ -166,72 +172,48 @@ def update_process_failure(self, planner: WorkPlanner,
-                                m0HaProcessEvent.M0_CONF_HA_PROCESS_STOPPED)
-                        elif current_status == ObjHealth.OK:
-                            proc_status = (
-                                m0HaProcessEvent.M0_CONF_HA_PROCESS_STARTED)
-                    if proc_status is not None:
-                        self.consul.update_process_status(
-                            ConfHaProcess(chp_event=proc_status,
-                                          chp_type=proc_type,
-                                          chp_pid=0,
-                                          fid=state.fid))
-                    # RC or not RC, i.e. even without persistent state update,
-                    # it is important that the notification to local motr
-                    # processes must still be sent.
-                    new_ha_states.append(
-                        HAState(fid=state.fid, status=current_status))
-                if not self.consul.is_proc_local(state.fid):
-                    proc_status_local = self.consul.get_process_status_local(
-                                            state.fid)
-                    # Consul monitors a process every 1 second and this
-                    # notification is sent to every node. Thus to avoid
-                    # notifying about a process multiple times about the
-                    # same status every node maintains a local copy of the
-                    # remote process status, which is checked everytime a
-                    # consul notification is received and accordingly
-                    # the status is notified locally to all the local motr
-                    # processes.
-                    if (current_status == ObjHealth.OK and
-                            (proc_status_local.proc_status !=
-                             'M0_CONF_HA_PROCESS_STARTED')):
-                        proc_status = (
-                            m0HaProcessEvent.M0_CONF_HA_PROCESS_STARTED)
-                    elif current_status in (ObjHealth.OFFLINE,
-                                            ObjHealth.FAILED):
-                        if (proc_status_local.proc_status !=
-                                'M0_CONF_HA_PROCESS_STOPPED'):
-                            proc_status = (
-                                m0HaProcessEvent.M0_CONF_HA_PROCESS_STOPPED)
-                    else:
-                        continue
-                    if proc_status is not None:
-                        self.consul.update_process_status_local(
-                            ConfHaProcess(chp_event=proc_status,
-                                          chp_type=proc_type,
-                                          chp_pid=0,
-                                          fid=state.fid))
+                proc_status = proc_Health_to_status.get(current_status)
+                if proc_status is not None:
+                    if proc_status_remote.proc_status != proc_status.name:
```

Comment:
It looks like we are comparing status with name.
@mssawant Is my understanding correct?

Reply:
proc_status.name is the enum.. so its name is OFFLINE or OK. (which is actually the status)

Reply:
`proc_status_remote` will contain a value as `processes/0x7200000000000001:0xa:{"state": "M0_CONF_HA_PROCESS_STARTED", "type": "M0_CONF_HA_PROCESS_M0D"}` so here `state` is `M0_CONF_HA_PROCESS_*` while current_status is of type `ObjHealth`, we map current status to `M0_CONF_HA_PROCESS_*`  using `proc_Health_to_status ` map. So we compare the process status saved in consul kv to it current expected status. If the saved status is different from the current expected status we update the same in consul kv, but only through RC.